lib/auth/auth.go

```diff
@@ -959,6 +964,86 @@ func (a *Server) syncUpgradeWindowStartHour(ctx context.Context) error {
 	return nil
 }
 
+func (a *Server) RetrieveExternalCloudAuditCredentials(ctx context.Context, integration string) (aws.Credentials, error) {
```

Comment:
I'd recommend rebasing on top of https://github.com/gravitational/teleport/pull/34380 where I've refactored the credential fetching a bit. Then you won't need this in your PR.

With that you can get credentials via:
```go
configurator, err := externalcloudaudit.NewDraftConfigurator(ctx, bk)
// handle err
configurator.SetGenerateOIDCTokenFn(a.GenerateExternalCloudAuditOIDCToken)
awsCfg, err := awsconfig.LoadDefaultConfig(ctx, awsconfig.WithCredentialsProvider(configurator.CredentialsProvider()))